### PR TITLE
apt: obtain supported suite directives for apt entitlements

### PR DESCRIPTION
**Pull request by blackboxsw**
ua-contracts service provides a 'suites' directive which can
be a list or None and describes the supported apt source suites
that can be enabled for a given entitlement.

Also adds default trusty-security and trusty-updates to package postinst.

Enable any supported suites that match the current running series

Fixes: #329
Fixes: #315

### CODE REVIEW

File: uaclient/apt.py [hunk 1]
```diff
@@ -66,12 +66,25 @@ def add_auth_apt_repo(repo_filename, repo_url, credentials, keyring_file=None,
...
+
     logging.info('Enabling authenticated repo: %s', repo_url)
     content = ''
-    for pocket in pockets:
-        content += ('deb {url}/ubuntu {series} {pocket}\n'
-                    '# deb-src {url}/ubuntu {series} {pocket}\n'.format(
-                        url=repo_url, series=series, pocket=pocket))
+    if not suites:
+        logging.debug('No apt source suite provided, assuming "%s".', series)
+        suites = [series]  # The default suite should be the series name
```
Comment (powersj) [inline uaclient/apt.py]: I debate whether suites should always be required, as we wouldn't ever have a xenial for esm

**Comment by powersj** [comment]
and one test error to fix:

```
>       assert apt_cmds == m_subp.call_args_list
E       AssertionError: assert [call(['apt-k...capture=True)] == [call(['apt-ca...capture=True)]
E         At index 0 diff: call(['apt-key', 'adv', '--keyserver', 'keyserver.ubuntu.com', '--recv-keys', 'APTKEY'], capture=True) != call(['apt-cache', 'policy'])
E         Right contains more items, first extra item: call(['apt-key', 'adv', '--keyserver', 'keyserver.ubuntu.com', '--recv-keys', 'APTKEY'], capture=True)
E         Use -v to get the full diff
uaclient/tests/test_apt.py:161: AssertionError
```

File: uaclient/apt.py (hunk 1 above)
Comment (blackboxsw) [reply uaclient/apt.py]: Sure let's complain loudly. I'm seeing  suites for esm, cc and fips defined by ua-contracts, so I'm good with that

File: debian/postinst [hunk 2]
```diff
@@ -8,10 +8,11 @@ configure_esm_source() {
     if [ ! -e "$ESM_APT_SOURCE_FILE" ]; then
         cat > $ESM_APT_SOURCE_FILE <<EOF
 # Written by ubuntu-advantage-tools during package postinst
```
Comment (powersj) [inline debian/postinst]: drop the "during package postinst"

File: debian/postinst [hunk 3]
```diff
@@ -8,10 +8,11 @@ configure_esm_source() {
...
-deb https://esm.ubuntu.com/ubuntu trusty security
-deb https://esm.ubuntu.com/ubuntu trusty updates
-# deb-src https://esm.ubuntu.com/ubuntu security
-# deb-src https://esm.ubuntu.com/ubuntu updates
+deb https://esm.ubuntu.com/ubuntu trusty-security main
+# deb-src https://esm.ubuntu.com/ubuntu trusty-security main
+
+deb https://esm.ubuntu.com/ubuntu trusty-updates
```
Comment (powersj) [inline debian/postinst]: missing main at the end

File: debian/postinst (hunk 2 above)
Comment (blackboxsw) [reply debian/postinst]: done.

**Pull request merged**